### ISSUE TITLE
Save Standalone Protect - Detect and Watermark if export is 60% identical to supplied demo game

### ISSUE THREAD

**Issue by LeeBamberTGC** [Dark-Basic-Software-Limited/GameGuruRepo#3414]
One of the major critiques in the Steam Universe was that some misguided users would export the demo games and sell them on Steam unchanged, causing quite a lot of bad feelings about the practice, from both Steam gamers and GameGuru creators.  One solution would be for the save standalone system to watermark any exports that the software has detected share 60% of it's DNA with one of the included game demo levels.  This would instantly prevent them from being sold on Steam, and would require the user to either create games from scratch or have them severely change an existing demo before the watermark disappears from the export.

Marked here as a feature request, but open to feedback on the wisdom and shortfalls of such a system. Not a problem most game makers have, but that's what you get when you make game making too easy ;)

**Comment by Necrym59** [comment]
Question: Why have standalone at all for a demo. Youre demoing the MAX editor not final exported product they can still play the demos so i would disable the export feature everyone gets the idea enough. Thats just my take on it.

**Comment by MonkeyFrogStudio** [comment]
I think that's a good point - the demos are there to learn from and be inspired from. They can be played by hitting the play button  at the beginning or by hitting the rocket icon once in the editor. If there were a way to restrict demos so they could not be exported as standalone, that would solve the problem, wouldn't it?

**Comment by earthling45** [comment]
Why not put a licence on those demo's as they are?

**Comment by fearlesswee** [comment]
I agree, the demos should probably just outright have exporting disabled. The 60% limit could easily be bypassed by bad actors by spamming crates outside of the map until it's just enough to be 40% different than the source demo level.

**Comment by Kitakazi** [comment]
Yea I agree, if someone still manages to upload a demo as a game on steam and takes the time to still get around the system, then file a copyright claim and press full legal charges.

**Comment by LeeBamberTGC** [comment]
Some very good ideas here. So we can make all demos none savable, both as standalones and as new game projects, that much is simple enough (much simpler than scanning for matches).  Is there a case where someone wants to start with a demo, then eventually call it their own and save standalone?  I cannot think of one, and we can pre-warn them to start a new game project to enable the save standalone feature :)

**Comment by MonkeyFrogStudio** [comment]
> Is there a case where someone wants to start with a demo, then eventually call it their own and save standalone?

I don't think so. But even so, the purpose of the demos is to 1) showcase what MAX can do and 2) help people learn to use the software by seeing how others have done things like connected visual logic nodes, etc. By not being able to start with a demo as a base, you are further encouraging people to create their own games from scratch from the very beginning, imo.
